src/Core/src/Hosting/MauiServiceProvider.cs

```diff
@@ -1,16 +1,21 @@
 using System;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace Microsoft.Maui.Hosting
 {
-
 	class MauiServiceProvider : IMauiServiceProvider
 	{
-		IDictionary<Type, Func<IServiceProvider, object?>?> _collection;
+		IMauiServiceCollection _collection;
+
+		// TODO: do this properly and support scopes
+		IDictionary<ServiceDescriptor, object?> _singletons;
```

Comment:
No lifetime management and might leak. I see the ASP.NET folks have a whole system in place, however since this is an embedded mobile app - things might not be too bad for now. But, looking at the ASP.NET system, there are more complex things to manage scopes and all of that. Once that gets done, this can go away for a real implementation.